src/inputdevice.cpp

```diff
@@ -601,10 +601,12 @@ void InputDevice::changeSetAxisButtonAssociation(int button_index, int axis_inde
     {
         button = getJoystick_sets().value(newset)->getJoyAxis(axis_index)->getPAxisButton();
     }
-
-    JoyButton::SetChangeCondition tempmode = static_cast<JoyButton::SetChangeCondition>(mode);
-    button->setChangeSetSelection(originset);
-    button->setChangeSetCondition(tempmode, true);
+    if (button != nullptr)
+    {
+        JoyButton::SetChangeCondition tempmode = static_cast<JoyButton::SetChangeCondition>(mode);
+        button->setChangeSetSelection(originset);
+        button->setChangeSetCondition(tempmode, true);
+    }
```

Comment:
This part is handled wrongly.    
Let's think about situation when button_index has invalid value (like 3)?     
What a good codebase should do in this case?

<details>
  <summary>Response</summary>
It should inform user/developer about this problem. You could use for this purpose either assert, warning or exception.

In case of this app warning should be the best option

```cpp
else{
WARN() << "Invalid button_index value: "<<button_index;
return;
}
```
  </details>